src/Foundation/NSObject2.cs

```diff
@@ -69,6 +69,15 @@ public class NSObjectFlag {
 	}
 #endif
 
+#if NET
+	public interface INSObjectFactory {
+		// The method will be implemented via custom linker step if the managed static registrar is used
+		// for NSObject subclasses which have an (NativeHandle) or (IntPtr) constructor.
+		[MethodImpl(MethodImplOptions.NoInlining)]
+		virtual static NSObject ConstructNSObject (NativeHandle handle) => null;
```

Comment:
I'm always concerned about name collisions. Are we OK with injecting a new method into classes that we don't control and can't ensure that this method doesn't already exist? Will this be handled by the `NewSlot` flag? I've not used it before.

Reply:
I haven't payed close attention to this problem so far. I've been trying to get it working first :) I definitely need to change something so that the new interfaces and methods aren't public and we don't introduce any new public APIs with this change.

I don't think this change can be implemented without adding the methods to the types that we want to create (especially because of generic types). We could in theory first check if there is a naming conflict and generate a different name if so. Or we could always add some random characters to decrease the likelihood that there will be a conflict.

Reply:
> Or we could always add some random characters to decrease the likelihood that there will be a conflict.

Adding a `_Xamarin_` prefix should be enough I think